src/rag_service.py

```
@@ -1,96 +1,128 @@
+        self.embedding_service = EmbeddingService()
 
-        except Exception as e:
-            logger.error("❌ Failed to initialize RAG Service: %s", e)
-            # We don't raise here to avoid blocking the bot, but RAG won't work
-            self._client = None
-
-    def add_documents(self, documents: List[str], metadatas: List[dict], ids: List[str]) -> bool:
-        """Add documents to the knowledge base."""
-        if not self._collection:
-            logger.warning("RAG Service not initialized, cannot add documents.")
+    async def add_documents(self, documents: list[str], metadatas: list[dict], ids: list[str]) -> bool:
```

Ruff (0.14.10)

Unused method argument: `ids`

(ARG002)